### PR TITLE
[33696] Retain autologin cookied request in session

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.com/wp/33696

### CODE REVIEW

File: app/controllers/account_controller.rb [hunk 1]
```diff
@@ -424,13 +424,14 @@ def successful_authentication(user, reset_stages: true)
   end
 
   def login_user!(user)
-    # Valid user
-    self.logged_user = user
     # generate a key and set cookie if autologin
-    if params[:autologin] && Setting.autologin?
+    if Setting.autologin? && (params[:autologin] || session[:autologin_requested])
       set_autologin_cookie(user)
```
Comment (machisuji) [inline app/controllers/account_controller.rb:429]: The `autologin_requested` value can be deleted after the cookie has been set, right?

File: app/services/projects/copy/work_packages_dependent_service.rb [hunk 2]
```diff
@@ -137,7 +137,6 @@ def copy_work_package_attribute_overrides(source_work_package, parent_id, user_c
         project: target,
         parent_id: parent_id,
         version_id: work_package_version_id(source_work_package),
-        assigned_to_id: work_package_assigned_to_id(source_work_package),
```
Comment (machisuji) [inline app/services/projects/copy/work_packages_dependent_service.rb]: Why was this removed? Is that wrong? Either way it has nothing to do with the autologin cookie. If it is a bug that you fixed while you were at it it would make sense to at least put it into a separate commit with a short explanation.

**Review by machisuji — CHANGES_REQUESTED** [review]
Looking good! Just two minor points.

File: app/services/projects/copy/work_packages_dependent_service.rb (hunk 2 above)
Comment (oliverguenther) [reply app/services/projects/copy/work_packages_dependent_service.rb]: Ah thanks, that was by mistake

File: app/controllers/account_controller.rb (hunk 1 above)
Comment (oliverguenther) [reply app/controllers/account_controller.rb:429]: Yes I'll just use `session.delete(:autologin_requested)`
